ui-core/src/components/inputs/datePicker/__tests__/useCalendar.spec.ts

```diff
@@ -0,0 +1,177 @@
+import { renderHook } from '@testing-library/react';
+import { describe, expect, it } from 'vitest';
+import useCalendar from '../useCalendar';
+
+describe('useCalendar', () => {
+  const february = 1;
+  const march = 2;
+  const november = 10;
+  const april = 3;
+  const july = 6;
+  const august = 7;
+  const june = 5;
+  const december = 11;
+
+  describe('day calculation', () => {
+    it('should include days from the previous month when the first day of the displayed month is not a Monday', () => {
+      const displayedMonthStartDate = new Date(2024, march, 1);
```

Comment:
Add a comment to specify which day it is (same for the other tests)
```suggestion
      const displayedMonthStartDate = new Date(2024, march, 1); // friday
```